sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,6 +20,226 @@
+    status_code : int, optional
+        HTTP status code indicating the result of the operation, defaults to 200.
+    body : str, optional
+        The content of the response, defaults to 'Success'.
+
+    Returns
+    -------
+    dict
+        A dictionary containing headers, status code, and body, designed to be returned
+        by a Lambda function as an API response.
+    """
+    if headers is None:
+        headers = (
+            {
+                "Content-Type": "text/html",
+            },
+        )
+    return {
+        "headers": headers,
+        "statusCode": status_code,
+        "body": body,
+    }
+
+
+def s3_event_handler(event):
+    """Handler function for S3 events.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+    # Retrieve the Object name
+    s3_filepath = event["detail"]["object"]["key"]
+
+    filename = os.path.basename(s3_filepath)
+    # TODO: add checks for SPICE or other
+    # data types
+
+    try:
+        science_file = ScienceFilepathManager(filename)
+    except InvalidScienceFileError as e:
+        logger.info(str(e))
+        return http_response(status_code=400, body=str(e))
+
+    # setup a dictionary of metadata parameters to unpack in the
+    # file catalog table
+    metadata_params = science_file.get_file_metadata_params()
+    metadata_params["file_path"] = s3_filepath
+
+    if metadata_params["data_level"] != "l0":
+        logger.info("Invalid data level. Expected l0")
+        return http_response(status_code=400, body="Invalid data level")
+
+    logger.info(f"Inserting {filename} into database")
+    # Add data to the file catalog and status tables
+    # create status params
+    # Event time looks like this:
+    # "time": "2024-01-16T17:35:08Z"
+    # Parses the time string from the event to a datetime object.
+    ingestion_date = datetime.datetime.strptime(event["time"], "%Y-%m-%dT%H:%M:%SZ")
+    # Formats the datetime object to a string with the format "%Y%m%d".
+    ingestion_data_str = ingestion_date.strftime("%Y%m%d")
+    # Parses the newly formatted string back into a datetime
+    # object with the desired structure.
+    ingestion_date_object = datetime.datetime.strptime(ingestion_data_str, "%Y%m%d")
+    status_params = {
+        "file_to_create_path": s3_filepath,
+        "status": models.Status.SUCCEEDED,
+        "job_definition": None,
+        "ingestion_date": ingestion_date_object,
+    }
+    try:
+        update_status_table(status_params)
+    except Exception as e:
+        logger.info(str(e))
+        return http_response(status_code=400, body=str(e))
+
+    # Query to get foreign key id for catalog table
+    status_tracking = None
+    with Session(db.get_engine()) as session:
+        # Query to get foreign key id for catalog table
+        status_tracking = (
```

Comment:
I'm still learning about sqlalchemy, but from what I saw in the documentation, it looked like the session.query().filter() is an older way of querying from version 1 and in version 2 it's more focused on select().where(), which is what I used in my query PR. If both are still valid, we should probably pick one way to be consistent unless there are differences between the two where one works better in certain situations that I"m not aware of.

Reply:
yes. I did look at your code was questioning myself too. I didn't read the document so what you are saying makes sense. I will make that change to be consistent.